addons/website/static/src/snippets/s_popup/options.js

```diff
@@ -61,13 +65,13 @@ options.registry.SnippetPopup = options.Class.extend({
      * @see this.selectClass for parameters
      */
     setLayout: function (previewMode, widgetValue, params) {
-        const isModal = widgetValue === 'modal';
-        const isTop = widgetValue === 'fixedTop';
-        this.$target.toggleClass('s_popup_fixed', !isModal);
-        this.$target.toggleClass('s_popup_fixed_top', isTop);
-        this.$target.toggleClass('s_popup_center modal', isModal);
-        this.$target.find('.s_popup_frame').toggleClass('modal-dialog modal-dialog-centered', isModal);
-        this.$target.find('.s_popup_content').toggleClass('modal-content', isModal);
+        const isTop = widgetValue === 'Top';
+        const isMiddle = widgetValue === 'Middle';
+        const isBottom = widgetValue === 'Bottom';
+        this.$target.toggleClass('s_popup_top', isTop);
+        this.$target.toggleClass('s_popup_middle', isMiddle);
+        this.$target.find('.modal-dialog').toggleClass('modal-dialog-centered', isMiddle);
+        this.$target.toggleClass('s_popup_bottom', isBottom);
```

Comment:
You can combine mulitple option methods and I would do it here. In xml, use data-select-class for the `s_popup_*` classes *and* data-set-layout to only update the modal-dialog element.